hack/gen-crd-spec/main.go

```diff
@@ -0,0 +1,121 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/argoproj/argo-cd/pkg/apis/application"
+	"github.com/argoproj/argo-cd/util/kube"
+
+	"github.com/ghodss/yaml"
+	extensionsobj "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+var (
+	kindToCRDPath = map[string]string{
+		application.ApplicationFullName: "manifests/crds/application-crd.yaml",
+		application.AppProjectFullName:  "manifests/crds/appproject-crd.yaml",
+	}
+)
+
+func getCustomResourceDefinitions() map[string]*extensionsobj.CustomResourceDefinition {
+	crdYamlBytes, err := exec.Command(
+		"controller-gen",
+		"paths=./pkg/apis/application/...",
+		"crd:trivialVersions=true",
+		"output:crd:stdout",
+	).Output()
+	checkErr(err)
+
+	// clean up stuff left by controller-gen
+	deleteFile("config/webhook/manifests.yaml")
+	deleteFile("config/webhook")
+	deleteFile("config")
+
+	objs, err := kube.SplitYAML(string(crdYamlBytes))
+	checkErr(err)
+	crds := make(map[string]*extensionsobj.CustomResourceDefinition)
+	for i := range objs {
+		un := objs[i]
+
+		// We need to completely remove validation of problematic fields such as creationTimestamp,
+		// which get marshalled to `null`, but are typed as as a `string` during Open API validation
+		removeValidataion(un, "metadata.creationTimestamp")
+		removeValidataion(un, "spec.template.metadata.creationTimestamp")
```

Comment:
We don't need `spec.template.metadata.creationTimestamp`

Reply:
done